src/Symfony/Component/Validator/Tests/Constraints/CollectionValidatorTest.php

```diff
@@ -386,4 +386,64 @@ public function testObjectShouldBeLeftUnchanged()
             'foo' => 3,
         ), (array) $value);
     }
+
+    /**
+     * @expectedException \Symfony\Component\Validator\Exception\ConstraintDefinitionException
+     */
+    public function testGroupShouldBePassedToTheContainingConstraint()
+    {
+        $value = array('baz' => 2);
+
+        $constraint = new Collection(
+             array(
+                 'fields' => array(
+                     'baz' => array(
+                         new Range(array('min' => 1, 'groups' => 'bar')),
+                     ),
+                 ),
+                 'groups' => 'foo',
```

Comment:
This test is wrong and breaks BC. Groups should be `array('foo,  'bar')` here, and the test would be green already, without the changes introduced in this PR.